cmd/sink/main.go

```diff
@@ -72,13 +75,56 @@ func NewSink(db database.DBInterface, filters *filters.Filters) *Sink {
 	}
 
 	return &Sink{
-		Db:          db,
-		EventClient: eventClient,
-		Filters:     filters,
-		K8sClient:   k8sClient,
+		Db:            db,
+		EventClient:   eventClient,
+		Filters:       filters,
+		K8sClient:     k8sClient,
+		logUrlBuilder: builder,
 	}
 }
 
+func (sink *Sink) writeLogs(ctx context.Context, obj *unstructured.Unstructured) {
+	kind := strings.ToLower(obj.GetKind())
+	if kind != "pod" {
+		slog.Info("Object does not have logs associated with it", "objectID", obj.GetUID(), "kind", kind)
+		return
+	}
+	logCtx, cancel := context.WithTimeout(ctx, time.Second*5)
+	defer cancel()
+	urls, err := sink.logUrlBuilder.Urls(logCtx, obj)
+	if err != nil {
+		slog.Error("Could build log urls for object", "objectID", obj.GetUID(), "err", err)
+		return
+	}
+	err = sink.Db.WriteUrls(logCtx, obj, urls...)
+	if err != nil {
+		slog.Error("Failed to write log urls for object to the database", "objectID", obj.GetUID(), "err", err)
+	} else {
+		slog.Info("Successfully wrote log urls for object to the database", "objectID", obj.GetUID(), "kind", kind)
+	}
+}
+
+func (sink *Sink) writeResource(ctx context.Context, obj *unstructured.Unstructured, event cloudevents.Event) {
+	dbCtx, cancel := context.WithTimeout(ctx, time.Second*5)
+	defer cancel()
+	err := sink.Db.WriteResource(dbCtx, obj, event.Data())
+	if err != nil {
+		slog.Error(
+			"Failed to write object from cloudevent to the database",
+			"objectID", obj.GetUID(),
+			"id", event.ID(),
+			"err", err,
+		)
+	} else {
+		slog.Info(
+			"Successfully wrote object from cloudevent to the database",
+			"objectID", obj.GetUID(),
+			"id", event.ID(),
+		)
+	}
+	sink.writeLogs(ctx, obj)
```

Comment:
Why call this method if you don't have a `Pod`?

Reply:
writeLogs is only called when `kind == Pod` now